packages/core/src/authenticatedFetch/dpopUtils.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import {
+  JWK,
+  KeyLike,
+  SignJWT,
+  generateKeyPair,
+  fromKeyLike,
+} from "@inrupt/jose-legacy-modules";
+import { v4 } from "uuid";
+
+/**
+ * Normalizes a URL in order to generate the DPoP token based on a consistent scheme.
+ *
+ * @param audience The URL to normalize.
+ * @returns The normalized URL as a string.
+ * @hidden
+ */
+function normalizeHttpUriClaim(audience: string): string {
```

Comment:
Personally, for such a generic function, I'd just name it more explicitly (i.e., the function itself has nothing to do with DPoP per se, or 'uri claims'). I would probably attempt to explain that it can be used, for example, to normalize URLs for reasons of comparability (at least as far as DPoP is concerned, perhaps for other uses too):
```suggestion
function removeHashUsernameAndPassword(url: string): string {
```